src/components/ApiPublishModal/ApiPublishModal.tsx

```diff
@@ -65,7 +63,7 @@ const ApiPublishModal: React.FC = () => {
   const [isApiMocked, setIsApiMocked] = useState<boolean>(false);
   const [isPublishDisabled, setIsPublishedDisabled] = useState<boolean>(true);
   const [redirectTabSelection, setRedirectTabSelection] = useState<string>('path_redirect');
-  const [upstreamRedirectTabSelection, setUpstreamRedirectTabSelection] = useState<string>('upstream');
+  const [targetSelection, setTargetSelection] = useState<string>('upstream');
   const [upstreamReference, setUpstreamReference] = useState<string>('service');
```

Comment:
It's not needed to type this useState. If you want - get rid of it. `string` type will be forced by typescript